website/pages/docs/connect/ingress_gateway.mdx

```diff
@@ -0,0 +1,77 @@
+  how to use Envoy and describes how you can plug in a gateway of your choice.
+---
+
+# Ingress Gateways
+
+-> **1.8.0+:**  This feature is available in Consul versions 1.8.0 and newer.
+
+Ingress gateways enable ingress traffic from services outside the Consul
+service mesh to services inside the Consul service mesh. An ingress gateway is
+a type of proxy and must be registered as a service in Consul, with the
+[kind](/api/agent/service#kind) set to "ingress-gateway". They are an
+entrypoint for outside traffic and allow you to define what services should be
+exposed and on what port. You configure an ingress gateway by defining a set of
+[listeners](/docs/agent/config-entries/ingress-gateway#listeners) that each map
+to a set of backing
+[services](/docs/agent/config-entries/ingress-gateway#services).
+
+Depending on the
+[protocol](/docs/agent/config-entries/ingress-gateway#protocol) defined for a
+listener, multiple services can be specified for a single listener. In this
+case, the ingress gateway relies on host/authority headers to decide the
+service that should receive the traffic.
+
+To enable easier service discovery, a new Consul [DNS
+subdomain](/docs/agent/dns#ingress-service-lookups) is provided, on
+`<service>.ingress.<domain>`.
+
+## Prerequisites
+
+Ingress gateways also require that your Consul datacenters are configured correctly:
+
+- You'll need to use Consul version 1.8.0.
+- Consul [Connect](/docs/agent/options#connect) must be enabled on the datacenter's Consul servers.
+- [gRPC](/docs/agent/options#grpc_port) must be enabled on all client agents.
+
+Currently, [Envoy](https://www.envoyproxy.io/) is the only proxy with ingress gateway capabilities in Consul.
+
+## Running and Using an Ingress Gateway
+
+You must complete the following steps to configure an ingress gateway to proxy traffic to services in the Consul service mesh:
+
+1. On a dedicated host for the gateway, start an Envoy proxy using the [envoy
+subcommand](/docs/commands/connect/envoy), specifying the `ingress` gateway
+type:
+    ```shell
+    $ consul connect envoy -gateway=ingress -register -service ingress-service \
+      -address '{{ GetInterfaceIP "eth0" }}:8888'
+    ```
+
+2. Create and apply an `ingress-gateway` [configuration entry](/docs/agent/config-entries/ingress-gateway) that defines
+a set of listeners that expose the desired backing services. The config entry can be applied via the
+[CLI](/docs/commands/config/write) or [API](/api/config#apply-configuration).
+
+3. Ensure that [Consul intentions](/docs/commands/intention) are setup between the ingress gateway and the backing services.
+
+4. **Optionally** use the `<service>.ingress.<domain>` [DNS subdomain](/docs/agent/dns#ingress-service-lookups) to discover the ingress
+gateways for a service.
+
+## Ingress Gateway Configuration
+
+Ingress gateways are defined similarly to other services registered with Consul, with two exceptions.
```

Comment:
I almost want to update this to say something like "Ingress gateways are defined in service configuration files and registered with Consul like other services, with two exceptions."

Reply:
I like it. Updated.

Reply:
Actually, thinking about this again, services don't need to be registered in files. So the original version is a bit more accurate.

Reply:
Could just remove the word "files", I think that would still make sense.

Reply:
Flipping this to be "Ingress gateways are configured in service definitions and registered with Consul like other services, with two exceptions."

Removing "files" since gateways are typically not going to be registered with files. The CLI command will auto-register them. Also the new wording mirrors the wording here: https://www.consul.io/docs/agent/services.html#service-definition